### ISSUE TITLE
Use PSR-18 "http-client" interface instead of Guzzle

### ISSUE THREAD

**Issue by kktsvetkov** [klaviyo/klaviyo-api-php#28]
This project is using Guzzle as its HTTP client. Nowadays most projects (including frameworks) already have a [PSR-18](https://github.com/php-fig/http-client) compatible HTTP client library. It's been like that for [some time](https://www.php-fig.org/blog/2018/11/psr-18-the-php-standard-for-http-clients/) and there are [options to choose from](https://packagist.org/providers/psr/http-client-implementation) if you want to use something different.

It will be a lot better if the selection of the HTTP client to use is left to the developer. You will remove the hard dependency to Guzzle, and make it easier to add test coverage for this project. 

You can still recommend Guzzle and its adapters in composer.json as your preferred suggestion. You can also add some very quick detection of known "http-client" projects to make it easier to use with less configuration.

**Comment by southcoast-matt** [comment]
I agree. I use symphony httpclient and didn't want guzzle installed!

**Comment by nickdnk** [comment]
I built this alternative. It supports PSR-18: https://github.com/nickdnk/klaviyo-php-sdk

Some other highlights: Properly typed input and output
Support for OAuth (aside from API keys) including rotation logic for your credential store
Webhook parsing (signature/drift verification) into typed resource
**A lot** smaller than this one
Fully unit tested

It's version 0.1 right now, so please give it a try and I'll release it under 1.0 in not too long.
